runtime/src/sorted_storages.rs

```diff
@@ -120,6 +134,66 @@ impl<'a> SortedStorages<'a> {
     }
 }
 
+/// Iterator over successive slots in 'storages' within 'range'.
+/// This enforces sequential access so that random access does not have to be implemented.
+/// Random access could be expensive with large sparse sets.
+pub struct SortedStoragesIter<'a> {
+    /// range for the iterator to iterate over (start_inclusive..end_exclusive)
+    range: Range<Slot>,
+    /// the data to return per slot
+    storages: &'a SortedStorages<'a>,
+    /// the slot to be returned the next time 'Next' is called
+    next_slot: Slot,
+}
+
+impl<'a> Iterator for SortedStoragesIter<'a> {
+    type Item = (Slot, Option<&'a SnapshotStorage>);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let slot = self.next_slot;
+        if slot < self.range.end {
```

Comment:
What about using `self.range.contains(slot)` here? Is it just to avoid checking `range.start`?

Reply:
Either could work. I was thinking since we know the iteration is always start..end, then slot is never < min. Seems like I could get rid of keeping range.start. Perhaps I simplify this later. Your reviews are always insightful.